presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewPlanValidator.java

```diff
@@ -36,6 +39,42 @@
     protected MaterializedViewPlanValidator()
     {}
 
+    @Override
+    protected Void visitQuerySpecification(QuerySpecification node, MaterializedViewPlanValidatorContext context)
+    {
+        process(node.getSelect(), context);
+        if (node.getFrom().isPresent()) {
+            process(node.getFrom().get(), context);
+        }
+        if (node.getWhere().isPresent()) {
+            Expression subquery = node.getWhere().get();
+            boolean wasWithinWhere = context.isWithinWhere();
+
+            context.setWithinWhere(true);
+            process(subquery, context);
+            context.setWithinWhere(wasWithinWhere);
+        }
+        if (node.getGroupBy().isPresent()) {
+            process(node.getGroupBy().get(), context);
+        }
+        if (node.getHaving().isPresent()) {
+            process(node.getHaving().get(), context);
+        }
+        if (node.getOrderBy().isPresent()) {
+            process(node.getOrderBy().get(), context);
+        }
+        return null;
+    }
+
+    @Override
+    protected Void visitTable(Table node, MaterializedViewPlanValidatorContext context)
+    {
+        if (context.isWithinWhere()) {
+            throw new SemanticException(NOT_SUPPORTED, node, "Subqueries inside where in materialized view are not supported yet");
+        }
+        return super.visitTable(node, context);
+    }
+
     public static void validate(Query viewQuery)
```

Comment:
Move this public interface to the top

Reply:
Please revise accordingly. Usually we use the order of public static -> public instance -> private static -> private instance.